components/org.wso2.carbon.siddhi.metrics.core/src/test/java/org/wso2/carbon/siddhi/metrics/core/StatisticsTestCase.java

```diff
@@ -0,0 +1,260 @@
+        memoryUsageTracker.registerObject(mockmoryObject,"test.memory");
+        AssertJUnit.assertEquals("test.memory",memoryUsageTracker.getName(mockmoryObject));
+        SiddhiBufferedEventsMetric  bufferedEventsTracker=(SiddhiBufferedEventsMetric)statisticsConfiguration
+                .getFactory().createBufferSizeTracker(new SiddhiStatisticsManager("MetricsTest",
+                        true));
+        EventBufferHolder eventBufferHolder = new EventBufferHolder() {
+            @Override
+            public long getBufferedEvents() {
+                return 1;
+            }
+
+            @Override
+            public boolean containsBufferedEvents() {
+                return true;
+            }
+        };
+       bufferedEventsTracker.registerEventBufferHolder(eventBufferHolder,"test.size");
+        AssertJUnit.assertEquals("test.size",bufferedEventsTracker.getName(eventBufferHolder));
+    }
+    @Test
+    public void statisticsTest1() throws InterruptedException {
+        log.info("statistics test 1");
+        SiddhiManager siddhiManager = new SiddhiManager();
+        StatisticsConfiguration statisticsConfiguration = new StatisticsConfiguration(new SiddhiMetricsFactory());
+        siddhiManager.setStatisticsConfiguration(statisticsConfiguration);
+        String siddhiApp = "@app:name('MetricsTest')" +
+                "@app:statistics(reporter = 'console', interval = '1' )" +
+                " " +
+                "define stream cseEventStream (symbol string, price float, volume int);" +
+                "define stream cseEventStream2 (symbol string, price float, volume int);" +
+                "" +
+                "@info(name = 'query1') " +
+                "from cseEventStream[70 > price] " +
+                "select * " +
+                "insert into outputStream ;" +
+                "" +
+                "@info(name = 'query2') " +
+                "from cseEventStream[volume > 90] " +
+                "select * " +
+                "insert into outputStream ;";
+
+        SiddhiAppRuntime siddhiAppRuntime = siddhiManager.createSiddhiAppRuntime(siddhiApp);
+        siddhiAppRuntime.addCallback("outputStream", new StreamCallback() {
+            @Override
+            public void receive(Event[] events) {
+                EventPrinter.print(events);
+                eventArrived = true;
+                for (Event event : events) {
+                    count++;
+                    AssertJUnit.assertTrue("IBM".equals(event.getData(0)) || "WSO2".equals(event.getData(0)));
+                }
+            }
+        });
+
+        InputHandler inputHandler = siddhiAppRuntime.getInputHandler("cseEventStream");
+        siddhiAppRuntime.start();
+        for (int i=0; i<1;i++) {
+            inputHandler.send(new Object[]{"WSO2", 55.6f, 100});
+            inputHandler.send(new Object[]{"IBM", 75.6f, 100});
+        }
+        Thread.sleep(3010);
```

Comment:
Why exactly 3010 millisec? Can we replace this with awaitility  lib?

Reply:
Fixed in b4a2bcf444d25031775a679bad76798d70ce7d5b